orquesta/conducting.py

```diff
@@ -882,3 +893,53 @@ def get_task_transition_contexts(self, task_id, route):
                 contexts[task_transition_id] = self.get_task_initial_context(t[1], route)
 
         return contexts
+
+    def request_workflow_rerun(self, options):
+        # Check current workflow status.
+        if self.get_workflow_status() not in statuses.ABENDED_STATUSES:
+            msg = 'Workflow execution cannot rerun from task(s) because it is not in a failed ' \
+                  'state'
+            raise exc.WorkflowInvalidRerunStatus(msg)
+
+        # Force reset workflow status to running
+        self.workflow_state.status = statuses.RESUMING
+
+        # Create a new entry in staging for the next task.
+        number_reran_tasks = 0
+        for task_id in options.get('tasks', []):
+            try:
+                task_route = 0
+                # Get latest task data
+                task = self.workflow_state.get_task(task_id, task_route)
+                if task and task['status'] in statuses.ABENDED_STATUSES:
+                    number_reran_tasks += 1
+                    ctxs = copy.deepcopy(task['ctxs']['in'])
+                    route = copy.deepcopy(task['route'])
+                    prev = copy.deepcopy(task['prev'])
+
+                    # Add a new task entry
+                    self.add_task_state(
+                        task_id,
+                        route,
+                        in_ctx_idxs=ctxs,
+                        prev=prev
+                    )
+
+                    # Create a new entry in staging for a failed task.
+                    self._workflow_state.add_staged_task(
+                        task_id,
+                        route,
+                        ctxs=ctxs,
+                        ready=True
+                    )
+            except exc.VariableInaccessibleError:
+                self.log_error("Couldn't find task %s in task table for rerun.", task_id)
+                continue
+            except Exception as e:
+                self.log_error(e)
+                continue
+
+        if number_reran_tasks == 0:
+            tasks_list = ', '.join(["%s" % t for t in options.get('tasks', [])])
+            self.workflow_state.status = statuses.FAILED
+            raise exc.InvalidTask(tasks_list)
```

Comment:
We don't need to raise an exception here.  All the errors are raised above and logged.  if we just failed the workflow here, the appropriate status and errors should be returned to st2.